packages/cli/src/environments/versionControl/versionControl.service.ee.ts

```diff
@@ -0,0 +1,75 @@
+import { Service } from 'typedi';
+import { generateSshKeyPair } from './versionControlHelper';
+import { VersionControlPreferences } from './types/versionControlPreferences';
+import { VERSION_CONTROL_PREFERENCES_DB_KEY } from './constants';
+import * as Db from '@/Db';
+import { jsonParse } from 'n8n-workflow';
+import type { Settings } from '@/databases/entities/Settings';
+
+@Service()
+export class VersionControlService {
+	private _versionControlPreferences: VersionControlPreferences = new VersionControlPreferences();
+
+	async init(): Promise<void> {
+		await this.loadFromDbAndApplyVersionControlPreferences();
+	}
+
+	public get versionControlPreferences(): VersionControlPreferences {
+		return {
+			...this._versionControlPreferences,
+			privateKey: '',
+		};
+	}
+
+	async generateAndSaveKeyPair(keyType: 'ed25519' | 'rsa' = 'ed25519') {
+		const keyPair = generateSshKeyPair(keyType);
+		if (keyPair.publicKey && keyPair.privateKey) {
+			this.setPreferences({ ...keyPair });
+			await this.saveSamlPreferencesToDb();
+		}
+		return keyPair;
+	}
+
+	setPreferences(prefs: Partial<VersionControlPreferences>) {
+		this._versionControlPreferences = {
+			...this._versionControlPreferences,
+			...prefs,
+		};
+	}
+
+	async loadFromDbAndApplyVersionControlPreferences(): Promise<
+		VersionControlPreferences | undefined
+	> {
+		const loadedPrefs = await Db.collections.Settings.findOne({
+			where: { key: VERSION_CONTROL_PREFERENCES_DB_KEY },
+		});
+		if (loadedPrefs) {
+			const prefs = jsonParse<VersionControlPreferences>(loadedPrefs.value);
+			if (prefs) {
+				this.setPreferences(prefs);
+				return prefs;
+			}
+		}
+		return;
+	}
+
+	async saveSamlPreferencesToDb(): Promise<VersionControlPreferences | undefined> {
+		const loadedPrefs = await Db.collections.Settings.findOne({
+			where: { key: VERSION_CONTROL_PREFERENCES_DB_KEY },
+		});
+		const settingsValue = JSON.stringify(this._versionControlPreferences);
+		let result: Settings;
+		if (loadedPrefs) {
+			loadedPrefs.value = settingsValue;
+			result = await Db.collections.Settings.save(loadedPrefs);
+		} else {
+			result = await Db.collections.Settings.save({
```

Comment:
I'm not 100% sure, but considering the `key` is the primary key, I guess typeorm can correctly insert or update by simply calling `save`